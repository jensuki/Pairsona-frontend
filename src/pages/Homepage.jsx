import { useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from '../context/UserContext';
import Loading from '../components/Loading';

// home images
import home1 from '../assets/homepage/home1.png';
import home2 from '../assets/homepage/home2.png';
import home3 from '../assets/homepage/home3.png';
import home4 from '../assets/homepage/home4.png';
import home5 from '../assets/homepage/home5.png';
import pairsonalogoLg from '../assets/logos/pairsona_logo_large.png';

import '../styles/Homepage.css';

const Homepage = () => {
    const { currUser, loading } = useContext(UserContext);

    if (loading) return <Loading />

    return (
        <div className="Homepage">
            {currUser ? (
                <>
                    {/* hero section for logged in users */}
                    <section className="hero" data-aos="fade-up">
                        <h3><i>Welcome {currUser.firstName}</i></h3>

                        {/* mbti introduction */}
                        <h1>Understanding MBTI Compatibility</h1>
                        <p>MBTI (Myers-Briggs Type Indicator) compatibility is rooted in the idea that personality types with complementary traits or cognitive functions can create harmonious and enriching relationships.
                            Compatibility doesn&apos;t necessarily mean “same type” — in fact, many MBTI pairings thrive on balance and contrast, where ones strengths complement another&apos;s.</p>
                        <p>It&apos;s often about balance. Opposites attract, like <strong>Introverts (I)</strong> pairing with <strong>Extraverts (E)</strong>, or <strong>Judgers (J)</strong> balancing with <strong>Perceivers (P)</strong>.
                            Cognitive functions also play a role — types with complementary strengths can create deep, meaningful connections.</p>
                        <Link to="/matches" className="cta-button">Get Matched</Link>

                        <hr></hr>

                        <h2>Examples of Compatible Pairings</h2>
                        <div className="example-pairs">
                            <div className="pair">
                                <img src={home1} alt="Girl smiling with book" />
                                <p><strong>INFJs</strong> bring deep introspection, while <strong>ENFPs</strong> add vibrant creative energy, balancing intuition and feeling.</p>
                            </div>
                            <div className="pair">
                                <img src={home2} alt="Male sitting with book" />
                                <p>Organized <strong>ISTJs</strong> pair well with lively and adaptable <strong>ESFPs</strong>.</p>
                            </div>
                            <div className="pair">
                                <img src={home3} alt="Gardening person" />
                                <p><strong>ENTPs</strong> bring fresh ideas, while <strong>INTJs</strong> provide focus and strategy.</p>
                            </div>
                            <div className="pair">
                                <p><strong>ISFJs</strong> offer stability, while <strong>ESTPs</strong> bring excitement and spontaneity.</p>
                            </div>
                            <div className="pair">
                                <img src={home4} alt="Girl running with flag" />
                                <p><strong>ENFJs</strong> guide with warmth and vision, while <strong>INFPs</strong> bring deep values and empathy.</p>
                            </div>
                            <div className="pair">
                                <img src={home5} alt="Male Logistician" />
                                <p><strong>ENTJs</strong> lead with ambition and drive, while <strong>INTPs</strong> offer insight and problem-solving.</p>
                            </div>
                        </div>
                    </section>

                </>
            ) : (
                <>
                    {/* hero section for guests */}
                    <section className="hero" data-aos="fade-up">
                        <h1>Authentic Connections<br></br> For Every Journey</h1>
                        <p>Meet like-minded people who align with your personality and values. Whether it&apos;s love, friendship, or meaningful chats, your connections - your way.</p>
                        <Link to="/quiz" className="cta-button">Take your MBTI Personality Quiz</Link>
                        <img className="logo-lg" src={pairsonalogoLg} alt="Guests Home avatar" />
                    </section>

                    <hr></hr>

                    {/* how it works section */}
                    <section className="how-it-works" data-aos="fade-left">
                        <h2>How It Works</h2>
                        <div className="steps">
                            <div className="step">
                                <h3>1. Sign Up</h3>
                                <p>Create an account and join our community.</p>
                            </div>
                            <div className="step">
                                <h3>2. Take the Quiz</h3>
                                <p>Discover your MBTI type with our simple quiz.</p>
                            </div>
                            <div className="step">
                                <h3>3. Get Matched</h3>
                                <p>Find compatible people based on personality.</p>
                            </div>
                            <div className="step">
                                <h3>4. Connect</h3>
                                <p>Connect and get to know your matches through our chat!</p>
                            </div>
                        </div>
                    </section>

                    <hr></hr>

                    {/* features section */}
                    <section className="features" data-aos="fade-up">
                        <h2>Why Choose Us?</h2>
                        <ul>
                            <li><strong>Personality-Based Matches</strong> – We pair you based on MBTI compatibility.</li>
                            <li><strong>Flexible Connections</strong> – Looking for friends or something more? Your choice.</li>
                            <li><strong>Privacy-Focused</strong> – Your data, your control.</li>
                        </ul>
                    </section>
                    <hr></hr>

                    {/* testimonials */}
                    <section className="testimonials">
                        <blockquote>This app made it easy to connect with fellow introverts in a discreet and comfortable way.</blockquote>
                        <blockquote>A welcoming space with plenty of opportunities to form genuine friendships.</blockquote>
                    </section>

                    {/* cta section */}
                    <section className="final-cta">
                        <h2>Ready to meet your match?</h2>
                        <Link to="/signup" className="cta-button">Get Started</Link>
                        <p className="login-link">
                            Already have an account? <Link to="/login">Log in</Link>
                        </p>
                    </section>
                </>
            )}
        </div>
    )
}

export default Homepage;