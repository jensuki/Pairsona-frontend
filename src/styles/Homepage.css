.Homepage {
    max-width: var(--max-width-lg);
}

.Homepage h3 {
    margin-top: 0;
}

.hero {
    margin-bottom: 2rem;
}

.hero h1 {
    font-family: 'Jost';
    font-size: 2.5rem;
    font-weight: normal;
    margin-bottom: 1rem;
}

.hero h2 {
    margin-top: 30px;
}

.hero p {
    text-align: left;
    font-size: 1.1rem;
    margin-bottom: 2rem;
}

/* examples of compatible pairings */

.example-pairs {
    display: flex;
    flex-direction: column;
    gap: 1em;
}

.pair {
    display: flex;
    align-items: center;
    gap: 1rem;
    height: 140px;
}

.pair:nth-child(odd),
.pair:nth-child(even) {
    border-radius: 300px 300px / 300px;
}

.pair:nth-child(odd) {
    flex-direction: row;
    box-shadow: 5px 0px 5px rgb(204, 198, 198);
}

.pair:nth-child(even) {
    flex-direction: row-reverse;
    box-shadow: -4px 2px 14px rgb(198, 194, 194);
}

.pair p {
    flex: .8;
}

.pair img {
    width: 200px;
    height: 200px;
}

/* cta buttons */

.cta-button {
    display: inline-block;
    background: var(--primary-green);
    color: var(--btn-white-text);
    padding: .75rem 2rem;
    border-radius: var(--border-radius-sm);
    font-weight: var(--font-bold);
    transition: var(--transition-bg);
    margin-bottom: 50px;

    &:hover {
        background-color: var(--hover-green);
    }
}

.final-cta {
    font-style: italic;
    background: #f5f5f5ab;
    border-radius: var(--border-radius-xl);
    padding: 2rem;
}

/* guests homepage */

.how-it-works {
    margin-bottom: 2rem;
}

.steps {
    display: flex;
    justify-content: center;
    gap: 2rem;
    flex-wrap: wrap;
}

.step {
    background: #f1f1f1;
    padding: 1rem;
    border-radius: var(--border-radius-xl);
    width: 250px;
}

.features {
    margin-bottom: 2rem;
}

.features ul {
    list-style: none;
}

.features li {
    background: #f1f1f1;
    padding: 1rem;
    margin: .5rem 0;
    border-radius: var(--border-radius-xl);
}

.testimonials {
    margin: 1rem 1rem;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;
    gap: .5rem;
    font-style: italic;
}

blockquote {
    max-width: 40%;
    min-width: 180px;
    margin: 1rem;
    font-family: Georgia, serif;
    font-style: italic;
    line-height: 1.3;
    position: relative;
    color: #383838;
}

blockquote:before {
    display: block;
    content: "\201c";
    font-size: 80px;
    position: absolute;
    left: -40px;
    top: -20px;
    color: #6b6b6b;
}

.logo-lg {
    box-shadow: none !important;
    height: auto;
    display: block;
    margin: 0 auto;
    padding-bottom: 2rem;
}