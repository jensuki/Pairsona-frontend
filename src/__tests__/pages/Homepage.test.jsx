import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../test_common/setupTests';
import Homepage from '../../pages/Homepage';
import { expect, describe, test } from 'vitest';
import { demoUser } from '../../test_common/mockData';

describe('Homepage component', () => {
    test('renders without crashing', () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <Homepage />
                </UserProvider>
            </MemoryRouter>
        );
    })

    test('matches snapshot for guests homepage', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider currUser={null}>
                    <Homepage />
                </UserProvider>
            </MemoryRouter>
        )
        expect(asFragment()).toMatchSnapshot();
    })

    test('matches snapshot for a logged in users homepage', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider currUser={demoUser}>
                    <Homepage />
                </UserProvider>
            </MemoryRouter>
        )

        expect(asFragment()).toMatchSnapshot();
    })

    test('renders home for anon users', () => {
        render(
            <MemoryRouter>
                <UserProvider currUser={null}>
                    <Homepage />
                </UserProvider>
            </MemoryRouter>
        );
        expect(screen.getByAltText(/Guests Home avatar/i)).toBeInTheDocument();
        expect(screen
            .getByRole('link', { name: /Take your MBTI Personality Quiz/i }))
            .toBeInTheDocument();
    })

    test('renders home for logged in users', () => {
        render(
            <MemoryRouter>
                <UserProvider currUser={demoUser}>
                    <Homepage />
                </UserProvider>
            </MemoryRouter>
        )
        expect(screen.getByText(/Welcome Test/i)).toBeInTheDocument();
        expect(screen.getByRole('link', { name: /Get Matched/i })).toBeInTheDocument()
    })
})