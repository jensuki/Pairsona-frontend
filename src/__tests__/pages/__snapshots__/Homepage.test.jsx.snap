// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage component > matches snapshot for a logged in users homepage 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <section
      class="hero"
      data-aos="fade-up"
    >
      <h3>
        <i>
          Welcome Test
        </i>
      </h3>
      <h1>
        Understanding MBTI Compatibility
      </h1>
      <p>
        MBTI (Myers-Briggs Type Indicator) compatibility is rooted in the idea that personality types with complementary traits or cognitive functions can create harmonious and enriching relationships. Compatibility doesn't necessarily mean “same type” — in fact, many MBTI pairings thrive on balance and contrast, where ones strengths complement another's.
      </p>
      <p>
        It's often about balance. Opposites attract, like 
        <strong>
          Introverts (I)
        </strong>
         pairing with 
        <strong>
          Extraverts (E)
        </strong>
        , or 
        <strong>
          Judgers (J)
        </strong>
         balancing with 
        <strong>
          Perceivers (P)
        </strong>
        . Cognitive functions also play a role — types with complementary strengths can create deep, meaningful connections.
      </p>
      <a
        class="cta-button"
        data-discover="true"
        href="/matches"
      >
        Get Matched
      </a>
      <hr />
      <h2>
        Examples of Compatible Pairings
      </h2>
      <div
        class="example-pairs"
      >
        <div
          class="pair"
        >
          <img
            alt="Girl smiling with book"
            src="/src/assets/homepage/home1.png"
          />
          <p>
            <strong>
              INFJs
            </strong>
             bring deep introspection, while 
            <strong>
              ENFPs
            </strong>
             add vibrant creative energy, balancing intuition and feeling.
          </p>
        </div>
        <div
          class="pair"
        >
          <img
            alt="Male sitting with book"
            src="/src/assets/homepage/home2.png"
          />
          <p>
            Organized 
            <strong>
              ISTJs
            </strong>
             pair well with lively and adaptable 
            <strong>
              ESFPs
            </strong>
            .
          </p>
        </div>
        <div
          class="pair"
        >
          <img
            alt="Gardening person"
            src="/src/assets/homepage/home3.png"
          />
          <p>
            <strong>
              ENTPs
            </strong>
             bring fresh ideas, while 
            <strong>
              INTJs
            </strong>
             provide focus and strategy.
          </p>
        </div>
        <div
          class="pair"
        >
          <p>
            <strong>
              ISFJs
            </strong>
             offer stability, while 
            <strong>
              ESTPs
            </strong>
             bring excitement and spontaneity.
          </p>
        </div>
        <div
          class="pair"
        >
          <img
            alt="Girl running with flag"
            src="/src/assets/homepage/home4.png"
          />
          <p>
            <strong>
              ENFJs
            </strong>
             guide with warmth and vision, while 
            <strong>
              INFPs
            </strong>
             bring deep values and empathy.
          </p>
        </div>
        <div
          class="pair"
        >
          <img
            alt="Male Logistician"
            src="/src/assets/homepage/home5.png"
          />
          <p>
            <strong>
              ENTJs
            </strong>
             lead with ambition and drive, while 
            <strong>
              INTPs
            </strong>
             offer insight and problem-solving.
          </p>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Homepage component > matches snapshot for guests homepage 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <section
      class="hero"
      data-aos="fade-up"
    >
      <h1>
        Authentic Connections
        <br />
         For Every Journey
      </h1>
      <p>
        Meet like-minded people who align with your personality and values. Whether it's love, friendship, or meaningful chats, your connections - your way.
      </p>
      <a
        class="cta-button"
        data-discover="true"
        href="/quiz"
      >
        Take your MBTI Personality Quiz
      </a>
      <img
        alt="Guests Home avatar"
        class="logo-lg"
        src="/src/assets/logos/pairsona_logo_large.png"
      />
    </section>
    <hr />
    <section
      class="how-it-works"
      data-aos="fade-left"
    >
      <h2>
        How It Works
      </h2>
      <div
        class="steps"
      >
        <div
          class="step"
        >
          <h3>
            1. Sign Up
          </h3>
          <p>
            Create an account and join our community.
          </p>
        </div>
        <div
          class="step"
        >
          <h3>
            2. Take the Quiz
          </h3>
          <p>
            Discover your MBTI type with our simple quiz.
          </p>
        </div>
        <div
          class="step"
        >
          <h3>
            3. Get Matched
          </h3>
          <p>
            Find compatible people based on personality.
          </p>
        </div>
        <div
          class="step"
        >
          <h3>
            4. Connect
          </h3>
          <p>
            Connect and get to know your matches through our chat!
          </p>
        </div>
      </div>
    </section>
    <hr />
    <section
      class="features"
      data-aos="fade-up"
    >
      <h2>
        Why Choose Us?
      </h2>
      <ul>
        <li>
          <strong>
            Personality-Based Matches
          </strong>
           – We pair you based on MBTI compatibility.
        </li>
        <li>
          <strong>
            Flexible Connections
          </strong>
           – Looking for friends or something more? Your choice.
        </li>
        <li>
          <strong>
            Privacy-Focused
          </strong>
           – Your data, your control.
        </li>
      </ul>
    </section>
    <hr />
    <section
      class="testimonials"
    >
      <blockquote>
        This app made it easy to connect with fellow introverts in a discreet and comfortable way.
      </blockquote>
      <blockquote>
        A welcoming space with plenty of opportunities to form genuine friendships.
      </blockquote>
    </section>
    <section
      class="final-cta"
    >
      <h2>
        Ready to meet your match?
      </h2>
      <a
        class="cta-button"
        data-discover="true"
        href="/signup"
      >
        Get Started
      </a>
      <p
        class="login-link"
      >
        Already have an account? 
        <a
          data-discover="true"
          href="/login"
        >
          Log in
        </a>
      </p>
    </section>
  </div>
</DocumentFragment>
`;
